diff --git a/src/Http/Middleware/ProcessIp.php b/src/Http/Middleware/ProcessIp.php
index 0a5ae94bf..7f4d6c5c6 100644
--- a/src/Http/Middleware/ProcessIp.php
+++ b/src/Http/Middleware/ProcessIp.php
@@ -19,7 +19,11 @@ class ProcessIp implements Middleware
 {
     public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
     {
-        $ipAddress = Arr::get($request->getServerParams(), 'REMOTE_ADDR', '127.0.0.1');
+        $params = $request->getServerParams();
+        $ipAddress = Arr::get($request->getServerParams(), 'REMOTE_ADDR', '127.0.0.1');
+        if ($cdnIpAddr = Arr::get($params, 'HTTP_X_FORWARDED_FOR')) {
+            $ipAddress = $cdnIpAddr;
+        }
 
         return $handler->handle($request->withAttribute('ipAddress', $ipAddress));
     }
--- a/src/Api/Serializer/AccessTokenSerializer.php
+++ b/src/Api/Serializer/AccessTokenSerializer.php
@@ -51,7 +51,7 @@ class AccessTokenSerializer extends AbstractSerializer
             'isCurrent' => $session && $session->get('access_token') === $token->token,
             'isSessionToken' => in_array($token->type, ['session', 'session_remember'], true),
             'title' => $token->title,
-            'lastIpAddress' => $token->last_ip_address,
+            'lastIpAddress' => explode(", ", "$token->last_ip_address")[0],
             'device' => $this->translator->trans('core.forum.security.browser_on_operating_system', [
                 'browser' => $agent->browser(),
                 'os' => $agent->platform(),
