diff --git a/less/forum.less b/less/forum.less
index f787836..bc3bbfb 100644
--- a/less/forum.less
+++ b/less/forum.less
@@ -1,4 +1,4 @@
-@import (css) "https://cdn.jsdelivr.net/npm/katex@0.13.11/dist/katex.min.css";
+@import (css) "https://lf6-cdn-tos.bytecdntp.com/cdn/expire-1-M/KaTeX/0.13.16/katex.min.css";
 
 @import "forum/copy-tex";
 @import "forum/text-editor";
diff --git a/src/ConfigureTextFormatter.php b/src/ConfigureTextFormatter.php
index b5d19c3..535b396 100644
--- a/src/ConfigureTextFormatter.php
+++ b/src/ConfigureTextFormatter.php
@@ -79,7 +79,7 @@ class ConfigureTextFormatter
                         <xsl:attribute name="data-s9e-livepreview-onrender">if(typeof katex!==\'undefined\')this.parentNode.removeChild(this)</xsl:attribute>
                         <xsl:attribute name="integrity">sha384-YNHdsYkH6gMx9y3mRkmcJ2mFUjTd0qNQQvY9VYZgQd7DcN7env35GzlmFaZ23JGp</xsl:attribute>
                         <xsl:attribute name="onload">katex.render(this.parentNode.innerText, this.parentNode, '.$options.')</xsl:attribute>
-                        <xsl:attribute name="src">https://cdn.jsdelivr.net/npm/katex@0.13.11/dist/katex.min.js</xsl:attribute>
+                        <xsl:attribute name="src">https://lf6-cdn-tos.bytecdntp.com/cdn/expire-1-M/KaTeX/0.13.16/katex.min.js</xsl:attribute>
                     </script>
                 </span>'
             );
