diff --git a/src/PushNotificationDriver.php b/src/PushNotificationDriver.php
index d9da4b6..e50af36 100644
--- a/src/PushNotificationDriver.php
+++ b/src/PushNotificationDriver.php
@@ -140,7 +140,7 @@ class PushNotificationDriver implements NotificationDriverInterface
          * @var MessageSentReport $report
          */
         foreach ($webPush->flush() as $report) {
-            if (!$report->isSuccess() && in_array($report->getResponse()->getStatusCode(), [401, 403, 404, 410])) {
+            if (!$report->isSuccess() && $report->getResponse() && in_array($report->getResponse()->getStatusCode(), [401, 403, 404, 410])) {
                 PushSubscription::where('endpoint', $report->getEndpoint())->delete();
             } elseif (!$report->isSuccess()) {
                 echo "[x] Message failed to sent for subscription {$report->getEndpoint()}: {$report->getReason()}";
