diff --git a/src/Providers/Provider.php b/src/Providers/Provider.php
index c09bb1a..4f4de15 100644
--- a/src/Providers/Provider.php
+++ b/src/Providers/Provider.php
@@ -35,7 +35,7 @@ class Provider extends AbstractServiceProvider
                 Discussion::class,
                 Page::class,
                 Tag::class,
-                User::class,
+                // User::class,
             ];
         });
 
