--- /dev/null
+++ ../src/Adapters/Manager.php
@@ -39,7 +39,8 @@
         protected Paths $paths,
         protected Util $util,
         protected SettingsRepositoryInterface $settings,
-        protected UrlGenerator $url
+        protected UrlGenerator $url,
+        protected Application $app,
     ) {
     }
 
@@ -107,7 +108,8 @@
 
         $leagueAdapter = new AwsS3Adapter(
             new S3Client($s3Config),
-            $this->settings->get('fof-upload.awsS3Bucket')
+            $this->settings->get('fof-upload.awsS3Bucket'),
+            $this->app->config('fof_upload_prefix', ''),
         );
 
         return new Adapters\AwsS3($leagueAdapter, $this->settings, $this->url);
