diff --git a/src/Adapters/Manager.php b/src/Adapters/Manager.php
index 9d7ed64..0e92620 100644
--- a/src/Adapters/Manager.php
+++ b/src/Adapters/Manager.php
@@ -37,8 +37,9 @@ class Manager
         protected Paths $paths,
         protected Util $util,
         protected SettingsRepositoryInterface $settings,
         protected UrlGenerator $url,
-        protected Config $config
+        protected Config $config,
+        protected \Flarum\Foundation\Application $app,
     ) {
     }
 
@@ -118,7 +119,8 @@ class Manager
 
         $leagueAdapter = new AwsS3Adapter(
             new S3Client($s3Config),
-            $config['bucket']
+            $config['bucket'],
+            $this->app->config('fof_upload_prefix', ''),
         );
 
         return new Adapters\AwsS3($leagueAdapter, $this->settings, $this->url, $this->config);
