diff --git a/helpers.php b/helpers.php
--- a/helpers.php
+++ b/helpers.php
@@ -181,7 +181,7 @@ if (! function_exists('optional')) {
      * @param  callable|null  $callback
      * @return mixed
      */
-    function optional($value = null, callable $callback = null)
+    function optional($value = null, ?callable $callback = null)
     {
         if (is_null($callback)) {
             return new Optional($value);
@@ -372,7 +372,7 @@ if (! function_exists('with')) {
      * @param  callable|null  $callback
      * @return mixed
      */
-    function with($value, callable $callback = null)
+    function with($value, ?callable $callback = null)
     {
         return is_null($callback) ? $value : $callback($value);
     }
