diff --git a/Container/Container.php b/Container/Container.php
--- a/Container/Container.php
+++ b/Container/Container.php
@@ -170,7 +170,7 @@ interface Container extends PsrContainer
      * @param  \Closure|null  $callback
      * @return void
      */
-    public function resolving($abstract, Closure $callback = null);
+    public function resolving($abstract, ?Closure $callback = null);
 
     /**
      * Register a new after resolving callback.
@@ -179,5 +179,5 @@ interface Container extends PsrContainer
      * @param  \Closure|null  $callback
      * @return void
      */
-    public function afterResolving($abstract, Closure $callback = null);
+    public function afterResolving($abstract, ?Closure $callback = null);
 }
